pkg/controller/perconaservermongodbrestore/physical.go

```diff
@@ -351,18 +364,17 @@ func (r *ReconcilePerconaServerMongoDBRestore) updateStatefulSetForPhysicalResto
 	)
 	sts.Spec.Template.Spec.InitContainers = append(sts.Spec.Template.Spec.InitContainers, pbmInit)
 
-	// remove backup-agent container
 	pbmIdx := -1
 	for idx, c := range sts.Spec.Template.Spec.Containers {
 		if c.Name == naming.ContainerBackupAgent {
 			pbmIdx = idx
 			break
 		}
 	}
-	if pbmIdx == -1 {
-		return errors.New("failed to find backup-agent container")
+	// remove backup-agent container
+	if pbmIdx != -1 {
+		sts.Spec.Template.Spec.Containers = append(sts.Spec.Template.Spec.Containers[:pbmIdx], sts.Spec.Template.Spec.Containers[pbmIdx+1:]...)
 	}
-	sts.Spec.Template.Spec.Containers = append(sts.Spec.Template.Spec.Containers[:pbmIdx], sts.Spec.Template.Spec.Containers[pbmIdx+1:]...)
```

Comment:
should we return error in the case when `pbmIdx == -1`?  it could be difficult to debug if we meet this problem.